utils/report_polling.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import sys
+import threading
+import time
+from http import HTTPStatus
+from utils.logger import Logger
+
+
+class ReportsConstants:
+    UFM_API_REPORTS = 'reports'
+
+
+class ReportPolling(object):
+    def __init__(self, ufm_rest_client):
+        self.ufm_rest_client = ufm_rest_client
+        self.action_inprogress = False
+
+    def start_polling(self, report_id):
+        try:
+            self.action_inprogress = True
+            t = self.create_loading_thread()
+            report_status = None
+            while report_status != HTTPStatus.OK:
+                time.sleep(3)
+                report_response = self.ufm_rest_client.send_request(f"{ReportsConstants.UFM_API_REPORTS}/{report_id}")
+                if report_response.raise_for_status():
```

Comment:
raise_for_status() will raise an exception in case the response status wasn't Ok
if you want to proceed with the code you need to catch & handle the raised expection